fili-core/src/test/groovy/com/yahoo/bard/webservice/web/SlicesApiRequestSpec.groovy

```diff
@@ -96,13 +107,17 @@ class SlicesApiRequestSpec extends Specification {
                 metricsResult.add(row)
             }
         }
+        Set<SortedMap<DateTime, Map<String, SegmentInfo>>> sliceMetadata = dataSourceMetadataService.getTableSegments(
+                Collections.singleton(table)
+        );
 
         Map<String, Object> expected = new LinkedHashMap<>()
         expected.put("name", name);
         expected.put("timeGrain", fullDictionary.get(name).getTimeGrain().getName());
         expected.put("timeZone", "UTC");
         expected.put("dimensions", dimensionsResult);
         expected.put("metrics", metricsResult);
+        expected.put("segmentInfo", SlicesApiRequest.generateSegmentMetadataView(sliceMetadata))
```

Comment:
Isn't this test supposed to be testing `SlicesApiRequest::generateSegmentMetadataView` (amongst other things)? As is, we're using the code under test to generate the expected value!